kiegroup/kie-wb-common
# JBPM-7145 Stunner: saving diagram SVG on backend

When a diagram is saved, the SVG referred to it should be generated and persisted on the backend, in the save way as the_ .bpmn2 _files. The SVG file should be tested integrating with the jbpm on the workbench.

It was created a command to handle the saving process `SaveDiagramSessionCommand`, for now it is handling the SVG saving after the Diagram itself is saved on the backend, the idea is to improve and handle all the saving logic within this command on a centralized way, we are on the half way on that and the rest of saving could be a separated JIRA. A refactor was necessary to support this modification, because the saving logic was spread when it is a the Project (integrated with WB) and when it is Standalone.

Other detail is that the SVG for BPMN2 need a specific key for the ids inserted on the SVG groups, so other logic were need to handle that, adding this key on the DefinitionSet in this way, other definition sets may have its own keys to identify the nodes when generating the SVG if necessary. Considering the SVG file name I implemented to have the same pattern as the old desginer the logic is on the `AbstractVFSDiagramService.getDiagramSvgFileName` that is **projectname. processname-svg.svg** .

**Steps to test:**

1. create a valid diagram without any bpmn violation and save it
2. build and deploy (on the authoring menu)
3. check the SVG generated on the WB, menu Manage -> Process Definitions -> select the created process.

This PR depends on appformer https://github.com/kiegroup/appformer/pull/337 and lienzo-core https://github.com/kiegroup/lienzo-core/pull/28.

@romartin 
@LuboTerifaj 

Guys this review I think is dedicated and you may have more information than me how to test the whole flow on the WB to make sure the SVG is generated on the proper way and is compatible and integrated with jpbm, the steps I inserted are what I used on my tests locally, besides checking the  **.niogit**  to view the bpmn and SVG created files.
